inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_async_request_thread_safe_default.hpp

```diff
@@ -245,32 +245,12 @@ class AsyncInferRequestThreadSafeDefault : public IAsyncInferRequestInternal {
         _syncRequest->SetBatch(batch);
     };
 
-    void GetUserData(void** data) override {
+    void SetCallback(Callback callback) override {
         CheckState();
-        if (data == nullptr) IE_THROW(NotAllocated);
-        *data = _userData;
+        _callback = std::move(callback);
     }
 
-    void SetUserData(void* data) override {
-        CheckState();
-        _userData = data;
-    }
-
-    void SetCompletionCallback(IInferRequest::CompletionCallback callback) override {
-        CheckState();
-        _callback = callback;
-    }
-
-    /**
-     * @brief Sets the pointer to public interface.
-     * @note Needed to correctly handle ownership between objects
-     * @param[in]  ptr A shared pointer to a public IInferRequest interface.
-     */
-    void SetPointerToPublicInterface(InferenceEngine::IInferRequest::Ptr ptr) {
-        _publicInterface = std::shared_ptr<IInferRequest>(ptr.get(), [](IInferRequest*) {});
-    }
-
-    std::vector<InferenceEngine::IVariableStateInternal::Ptr> QueryState() override {
+    std::vector<std::shared_ptr<InferenceEngine::IVariableStateInternal>> QueryState() override {
```

Comment:
should not we call `CheckState();` here as well?

Reply:
added